lib/ansible/modules/cloud/huawei/hwc_vpc_route.py

```diff
@@ -0,0 +1,436 @@
+# create a peering connect
+- name: create a local vpc
+  hwc_network_vpc:
+    cidr: "192.168.0.0/16"
+    name: "ansible_network_vpc_test_local"
+  register: vpc1
+- name: create a peering vpc
+  hwc_network_vpc:
+    cidr: "192.168.0.0/16"
+    name: "ansible_network_vpc_test_peering"
+  register: vpc2
+- name: create a peering connect
+  hwc_vpc_peering_connect:
+    local_vpc_id: "{{ vpc1.id }}"
+    name: "ansible_network_peering_test"
+    filters:
+      - "name"
+    peering_vpc:
+      vpc_id: "{{ vpc2.id }}"
+  register: connect
+- name: create a route
+  hwc_vpc_route:
+    vpc_id: "{{ vpc1.id }}"
+    destination: "192.168.0.0/16"
+    next_hop: "{{ connect.id }}"
+    filters:
+      - "vpc_id"
+      - "destination"
+      - "type"
+'''
+
+RETURN = '''
+    destination:
+        description:
+            - Specifies the destination IP address or CIDR block.
+        type: str
+        returned: success
+    next_hop:
+        description:
+            - Specifies the next hop. The value is VPC peering connection ID.
+        type: str
+        returned: success
+    vpc_id:
+        description:
+            - Specifies the VPC ID to which route is added.
+        type: str
+        returned: success
+    type:
+        description:
+            - Specifies the type of route.
+        type: str
+        returned: success
+'''
+
+from ansible.module_utils.hwc_utils import (
+    Config, HwcClientException, HwcModule, are_different_dicts, build_path,
+    get_region, is_empty_value, navigate_value)
+
+
+def build_module():
+    return HwcModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'],
+                       type='str'),
+            filters=dict(required=True, type='list', elements='str'),
+            destination=dict(type='str', required=True),
+            next_hop=dict(type='str', required=True),
+            vpc_id=dict(type='str', required=True),
+            type=dict(type='str', default='peering')
+        ),
+        supports_check_mode=True,
+    )
+
+
+def main():
+    """Main function"""
+
+    module = build_module()
+    config = Config(module, "vpc")
+
+    try:
+        resource = None
+        if module.params['id']:
+            resource = True
+        else:
+            v = search_resource(config)
+            if len(v) > 1:
+                raise Exception("find more than one resources(%s)" % ", ".join([
+                                navigate_value(i, ["id"]) for i in v]))
+
+            if len(v) == 1:
+                resource = v[0]
+                module.params['id'] = navigate_value(resource, ["id"])
+
+        result = {}
+        changed = False
+        if module.params['state'] == 'present':
+            if resource is None:
+                if not module.check_mode:
+                    create(config)
+                changed = True
+
+            result = read_resource(config)
+            result['id'] = module.params.get('id')
```

Comment:
Is there any modification code? I.e. if the resource already exists, but is different, will the module modify it?

Reply:
There isn't any modification code,  we don't support modify this module.

Reply:
Users in general expect Ansible modules to be idempotent. There are usually two solutions for this:
 a) document that it is not idempotent;
 b) make it idempotent by deleting the old object and recreating it with the new parameters if you see that parameters changed.

Obviously, solution b) is not always a good idea (like when user data is deleted). In cases where recreation can destroy things, I would allow the user to tell the module what it should do (like a choices option with values `ignore`, `recreate`, `fail` - that would also be easy to extend in case a later API allows to modify the configuration).

Reply:
Hi felixfontein, Thanks for reviewing the code.
The route has been search by all parameters(destination, next_hop, vpc_id, type). 

If we don't find the route resource, we will create a new one with new parameters.

If we find the route resource(code: "v = search_resource(config)" ), it means this route resource attributes(destination, next_hop, vpc_id, type) is the same as the parameters inpupt by the user(for example: route.yml). 

So it is idempotent.

Reply:
If all these options identify the object, then it's fine.

Reply:
What happens if `id` is specified, but these options do not match to the object identified by `id`?

Reply:
Good Point!  Thank you to point it out!! I will fix it ASAP.

Reply:
The "id" parameter don't support as the input paramter, we can only get "id" in retrun values. So the "id" cannot be specify after I fix it.

Reply:
Done